File: spec/Insertion_Sort_Spec.rb
require 'spec_helper'

describe 'Insertion Sort' do

  describe 'sort' do
    it 'should sort the collection from small to large' do
      is = Insertion_sort.new
      container = My_Container.new [9,8,7,6,5,4,3,2,1]
      sorted_container = My_Container.new [1,2,3,4,5,6,7,8,9]

      is.sort container
      container.equal(sorted_container).should be_true


      container = My_Container.new [5,7,2,9,6,1,1]
      sorted_container = My_Container.new [1,1,2,5,6,7,9]

      is.sort container
      container.equal(sorted_container).should be_true


      container = My_Container.new [98]
      sorted_container = My_Container.new [98]

      is.sort container
      container.equal(sorted_container).should be_true
    end
  end
end

File: spec/javascripts/InsertionSortSpec.js
describe('InsertionSort', function() {

    describe('sort', function() {
        var is, container;
        beforeEach(function() {
            is = new InsertionSort();
            container = new MyContainer([9,8,7,6,5,4,3,2,1,0]);
            sortedContainer = new MyContainer(10);
        });

        it('should sort the container', function() {
            expect(is.sort(container)).toEqual(sortedContainer);

            container = new MyContainer([9,3,2,4,1,5,6,0,7,8]);
            expect(is.sort(container)).toEqual(sortedContainer);
        });
    });
});
